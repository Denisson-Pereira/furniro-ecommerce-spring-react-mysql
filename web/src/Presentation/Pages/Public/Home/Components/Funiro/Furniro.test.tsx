import { render, screen } from "@testing-library/react";
import { I18nextProvider } from "react-i18next";
import { describe, it, expect } from "vitest";
import i18n from "../../../../../Translate";
import { Funiro } from ".";
import { MemoryRouter } from "react-router-dom";

describe("Funiro Component", () => {
    it("should render Funiro component with translated text", () => {
        const { container } = render(
            <MemoryRouter>
                <I18nextProvider i18n={i18n}>
                    <Funiro />
                </I18nextProvider>
            </MemoryRouter>
        );

        expect(screen.getByText(i18n.t("home.title4"))).toBeInTheDocument();

        expect(screen.getByText("#FuniroFurniture")).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
