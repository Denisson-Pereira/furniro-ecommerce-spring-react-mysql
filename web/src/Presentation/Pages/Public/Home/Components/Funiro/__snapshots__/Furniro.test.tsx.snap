// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Funiro Component > should render Funiro component with translated text 1`] = `
<div>
  <div
    class="funiro_container"
  >
    <div
      class="funiro_title"
    >
      <p>
        Share your setup with
      </p>
      <span>
        #FuniroFurniture
      </span>
    </div>
  </div>
</div>
`;
